src/groups/mqb/mqbc/mqbc_clusterutil.cpp

```diff
@@ -897,6 +896,20 @@ ClusterUtil::assignQueue(ClusterState*         clusterState,
             allocator);
         domIt = clusterState->domainStates().find(uri.qualifiedDomain());
     }
+    else {
+        // Set the queue as assigning (no longer pending unassignment)
+
+        UriToQueueInfoMapCIter qcit = domIt->second->queuesInfo().find(uri);
+        if (qcit != domIt->second->queuesInfo().cend()) {
```

Comment:
This approach doesn't actually work. Assignment is a different story than unassignment. During first pending assignment, the queue does not exist in `ClusterState` yet, so `qcit` won't find anything, and thus won't set anything to `k_ASSIGNING`. 

*Lemma*: If the queue exists in `ClusterState`, then it is never in `k_ASSIGNING` state.

Therefore, we need a separate collection to keep track of `k_ASSIGNING` queues, and once assignment at `ClusterState` completes, we need to remove the corresponding entry.